# Malawi Course repo

Repository to hold notes, files and links for the Malawi training.

Currently private but we will make this public so participants can download files,
find links, reread instructions, etc.

Home: <https://github.com/seqafrica/malawi-course>


## Programme

 * Google doc: <https://docs.google.com/document/d/1oN3s6jiw1a7ccOcbP2N_ZHP_qV_GEZgAjZKRn4Mt56M/edit?tab=t.0>


## TODO: ONT blog on Nanopore Sequencing, very insightful

<https://nanoporetech.com/blog/how-oxford-nanopore-sequencing-works> has
very nice animation <https://youtu.be/RcP85JHLmnI>.


## TODO: InterpretAMR / AMRgen, AMRrules

<https://github.com/interpretAMR>

**AMRgen**:

AMRgen is an open-source R package designed to bridge the gap between genotypic and phenotypic antimicrobial resistance (AMR) data. Developed as an extension to the AMR R package, it provides tools to interpret AMR genes, integrate these findings with antimicrobial susceptibility test (AST) data, and calculate genotype-phenotype associations.

This package is developed in collaboration with the ESGEM-AMR Working Group and is tailored for researchers and healthcare professionals tackling AMR globally.

Also:

From their github page:

We are partnering with ESGEM, the ESCMID Study Group on Epidemiological Markers, to form an ESGEM-AMR Working Group to curate organism-specific rule sets. Membership of the working group is open to anyone with relevant expertise, you do not have to be an ESGEM member.

If you would like to get involved, please:

    Consider registering to attend one of the intro sessions to find out if this is the group for you:

    Tues 14 May 17:00 UK / 9:00 Seattle / 12:00 Boston
    Weds 15 May 9:00 UK / 15:00 Hanoi / 18:00 Melbourne

    If you decide you want to join the ESGEM-AMR working group please register your interest, and let us know what organism/s you have expertise in, using this form.

 
## Enterobase

Explore any dataset (for the nine species that Enterobase covers) that is publicly
available in the SRA and/or your own uploaded isolates.  Illumina short reads only.

Enterobase has GrapeTree built in, but you can also use it standalone.

#### Links:

 * Site: <https://enterobase.warwick.ac.uk/>
 * Documentation: <https://enterobase.readthedocs.io/en/latest/>


## Grapetree

See [grapetree/README.md](grapetree/README.md)

