# Malawi Training 4-7 March 2025

Repository to hold notes, files and links for the Malawi training

Home: <https://github.com/seqafrica/malawi-course>


## Programme

 * [Malawi training agenda](Malawi_training_agenda_final.pdf)

## Day 1

Slides

 * [Bioinformatics Basics](https://zwets.it/course/malawi/slides/bioinformatics-basics.pdf)
 * [BAP Slides](https://zwets.it/course/malawi/slides/bap-slides.pdf)

Links
 
 * [Blog explaining ONT sequencing](https://nanoporetech.com/blog/how-oxford-nanopore-sequencing-works)
 * [Nice animation of ONT sequencing](https://youtu.be/RcP85JHLmnI)

Exercise links

 * [Mystery species (did we really find Asgard archaea?)](mystery/README.md)
 * [Bacterial Species identification](identify/README.md)

## Day 2

Slides

 * [Illumina Reads QC](https://zwets.it/course/malawi/slides/illumina-reads-qc.pdf)
 * [FastQC Puzzles](https://zwets.it/course/malawi/slides/fastqc-puzzles.pdf)

Links

 * [Illumina Sequencing by Synthesis video](https://www.youtube.com/watch?v=fCd6B5HRaZ8)
 * [Land Rover Sequencing Report](https://zwets.it/course/malawi/landrover/report.pdf)

Exercise links

 * (Optional: for command-line experimenters) [Installing WSL and EPI2ME on Windows](https://zwets.it/course/malawi/epi2me/windows-epi2me-install.pdf)
 * [FastQC analysis of MiSeq reads](fastqc/README.md)
 * [Interpret QC analysis of Malawi data](qc-mw-data/README.md)

## Day 3

Slides

 * [Bioinformatics Recap](https://zwets.it/course/malawi/slides/bioinformatics-recap.pdf)

Links

 * [hAMRonization](https://github.com/pha4ge/hAMRonization)

Exercise links

 * [Build your own anything-finder](mydbfinder/README.md)
 * [AMR detection with online tools](amr/README.md)

## Day 4

Links

 * Enterobase <https://enterobase.warwick.ac.uk/>
 * Enterobase documentation: <https://enterobase.readthedocs.io/en/latest/>

Exercise links

 * [Grapetree](grapetree/README.md)

